### ISSUE TITLE
Inconsistency in the training log with YOLO11n-cls

### ISSUE THREAD

**Issue by brenoslinhares** [ultralytics/ultralytics#22670]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

The following information is appearing in the training log for the YOLO11n-cls model:

[...]
**Overriding model.yaml nc=80 with nc=2**
[...]

Upon testing, I verified that the nc of the .yaml file extracted from the pre-trained YOLO11n-cls model is 80, as shown in the image below:

<img width="907" height="492" alt="Image" src="https://github.com/user-attachments/assets/06813b1d-edc8-49a5-8b72-fb1b98f07860" />

However, the [website](https://docs.ultralytics.com/models/yolo11/#classification-imagenet) states that YOLO11 for the classification task was pre-trained on the ImageNet-1K dataset.

This inconsistency does not appear with the YOLOv8n-cls model, as shown in the test below:

<img width="907" height="492" alt="Image" src="https://github.com/user-attachments/assets/27dea0d4-edc5-4607-b1ba-8b0cad7a0057" />

Was the YOLO11n-cls model pre-trained on COCO-80 or ImageNet-1K?

Thank you in advance.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @brenoslinhares, thank you for your interest in Ultralytics 🚀! This is an automated response to help get you unblocked quickly. An Ultralytics engineer will also review your issue and assist soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including:
-Exact CLI/Python commands used
-Full training logs showing the “Overriding model.yaml nc=...” line
-ultralytics package version and environment details (OS, Python, PyTorch, CUDA if applicable)
-A minimal snippet or reference to the specific model file you’re loading (e.g., YOLO11n-cls) and how you load it
And verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). Sharing these details will help us quickly understand the log behavior you’re observing 🔎

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
This is the `nc` that was defined in the YAML used for training. But that doesn't mean the model was trained on COCO. The actual `nc` is based on the dataset which  is 1000 for YOLO11 classification models

**Comment by brenoslinhares** [comment]
@Y-T-G,  but the `nc` defined in the `.yaml` file is 1000, not 80. Where in the code does this change to 80 occur, specifically for the `YOLO11n-cls`?

<img width="922" height="826" alt="Image" src="https://github.com/user-attachments/assets/6e094a0c-1f23-4107-aa2a-2a44bbb7dcf1" />

**Comment by Y-T-G** [comment]
That's YAML in repo. The model checkpoint also stores a copy of YAML. In case of YOLO11n-cls, the YAML that was used during training had `nc` as 80 which is stored in the model checkpoint. When you start training using a pretrained model, it uses the YAML in the model, not the one in the repo

**Comment by brenoslinhares** [comment]
@Y-T-G , So, to confirm: the class count (nc) used in training the YOLOv8n-cls model was 1000 with an ImageNet-1K dataset, and the class count for the YOLO11n-cls model was 80, also with ImageNet-1K. The difference in the number of classes used was smaller for YOLO11.

**Comment by Y-T-G** [comment]
No. The `nc` in the model YAML is overridden by the dataset `nc`. Both YOLOv8n-cls and YOLO11n-cls have 1000 classes. Because that's the number of classes in ImageNet. The `nc` in the model YAML doesn't affect that.
